src/runtime/proxy.ts

```diff
@@ -1,17 +1,116 @@
 import { Context } from "../runtime/mod.ts";
 import { RequestOf, ResponseOf } from "../types/registry.ts";
 
+/**
+ * This type is a union of all of the valid script names for a SPECIFIC module
+ * in a registry.
+ */
 type AnyScriptName<RegistryT, Key extends keyof RegistryT> = keyof RegistryT[Key];
+
+/**
+ * This type is a union of ALL valid [<module>, <script-in-module>] pairs in a
+ * registry.
+ * 
+ * This is used primarily in {@link MapFrom} to denote a valid mapping into a
+ * registry.
+ */
 type AnyPair<RegistryT> = {
 	[Key in keyof RegistryT]: readonly [Key, AnyScriptName<RegistryT, Key>];
 }[keyof RegistryT];
 
+/**
+ * This type is used denote a map the key/value pairs of one registry to
+ * another.
+ * 
+ * 
+ * Example:
+ * ```ts
+ * type TestReg1 = {
+ *   foo: {
+ *     bar: { request: FooBarRequest, response: FooBarResponse },
+ *     baz: { request: FooBazRequest, response: FooBazResponse },
+ *   },
+ *   fil: {
+ *     qux: { request: FilQuxRequest, response: FilQuxResponse },
+ *     cor: { request: FilCorRequest, response: FilCorResponse },
+ *   }
+ * };
+ * 
+ * type TestReg2 = {
+ *   canonicalFoo: {
+ *     bar: { request: FooBarRequest, response: FooBarResponse },
+ *     bazScript: { request: FooBazRequest, response: FooBazResponse },
+ *   },
+ *   filMod: {
+ *     qux: { request: FilQuxRequest, response: FilQuxResponse },
+ *     cor: { request: FilCorRequest, response: FilCorResponse },
+ *   },
+ * };
+ * 
+ * const map: MapFrom<TestReg1, TestReg2> = {
+ *   foo: {
+ *     bar: ["canonicalFoo", "bar"],
+ *     baz: ["canonicalFoo", "bazScript"],
+ *   },
+ *   fil: {
+ *     qux: ["filMod", "qux"],
+ *     cor: ["filMod", "cor"],
+ *   },
+ * };
```

Comment:
It would be nice to include an example of what our code is doing between the camel case registry and snake case registry